aten/src/ATen/core/ivalue.h

```diff
@@ -324,6 +324,7 @@ struct CAFFE2_API IValue final {
   // ConstantString
   IValue(c10::intrusive_ptr<ivalue::ConstantString> v);
   IValue(std::string v);
+  IValue(const char* v): IValue(std::string(v)) {}
```

Comment:
One idea - what if we explicitly banned all direct conversions? I'm not sure whether it's a good idea or not:

```
  template<class T> IValue(T x) {
      static_assert(!std::is_same<T,T>::value, "Implicit conversions are disabled for IValues to avoid accidental casting to bool");
  }
```

It'd require though adding more variants of constructors as some of the constness conversions might be disabled too.

Reply:
sounds like a good idea